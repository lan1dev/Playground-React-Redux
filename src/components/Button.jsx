import styled from 'styled-components';

const Button = styled.button`
  border: #b2dfdb;
  background: ${props => (props.error ? '#dd2c00' : '#00695c')};
  color: ${props => (props.error ? '#FFAB91' : '#b2dfdb')};
  height: 50px;
  width: 100px;
  font-size: 18px;
  margin: 0 10px;
  transition: background 0.3s;
  &:hover {
    transition: background 0.3s;
    background: ${props => (props.error ? '#BF360C' : '#004d40')};
  }
`;

export default Button;
