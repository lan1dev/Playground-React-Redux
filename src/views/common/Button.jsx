import styled from 'styled-components';
import colors from '../constants/colors';

const Button = styled.button`
  border: #b2dfdb;
  background: ${props => (props.error ? colors.error : colors.primary)};
  color: ${props => (props.error ? colors.lightError : colors.lightPrimary)};
  height: 50px;
  width: 100px;
  font-size: 18px;
  margin: 0 10px;
  transition: background 0.3s;
  &:hover {
    transition: background 0.3s;
    background: ${props => (props.error ? colors.darkError : colors.darkPrimary)};
  }
`;

export default Button;
