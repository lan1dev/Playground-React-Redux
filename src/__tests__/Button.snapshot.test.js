import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import Button from '../components/Button';

describe('Button', () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<Button {...props}>{props.children}</Button>);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {
      children: undefined,
      error: undefined
    };
    wrapper = undefined;
  });

  describe('when props are undefined', () => {
    it('Should match snapshot', () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });

    it('should not have green background', () => {
      const tree = Wrapper();
      expect(tree).toHaveStyleRule('background', '#00695c');
    });

    it('should not have green color', () => {
      const tree = Wrapper();
      expect(tree).toHaveStyleRule('color', '#b2dfdb');
    });
  });

  describe('when props are defined', () => {
    it('should match snapshot', () => {
      props.children = 'Text';
      props.error = true;
      expect(toJson(Wrapper())).toMatchSnapshot();
    });

    it('should be have red background', () => {
      props.children = 'Text';
      props.error = true;
      expect(Wrapper()).toHaveStyleRule('background', '#dd2c00');
    });
    it('should be have red color', () => {
      props.children = 'Text';
      props.error = true;
      expect(Wrapper()).toHaveStyleRule('color', '#FFAB91');
    });
    describe('user hovers over the button', () => {
      it('should have darkred background', () => {
        props.children = 'Text';
        props.error = true;
        wrapper = Wrapper();
        expect(wrapper).toHaveStyleRule('background', '#BF360C', {
          modifier: ':hover'
        });
      });
    });
  });
});
