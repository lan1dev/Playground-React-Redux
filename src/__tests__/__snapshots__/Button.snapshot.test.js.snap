// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button when props are defined should match snapshot 1`] = `
.c0 {
  border: #b2dfdb;
  background: #dd2c00;
  color: #FFAB91;
  height: 50px;
  width: 100px;
  font-size: 18px;
  margin: 0 10px;
  -webkit-transition: background 0.3s;
  transition: background 0.3s;
}

.c0:hover {
  -webkit-transition: background 0.3s;
  transition: background 0.3s;
  background: #BF360C;
}

<button
  className="c0"
>
  Text
</button>
`;

exports[`Button when props are undefined Should match snapshot 1`] = `
.c0 {
  border: #b2dfdb;
  background: #00695c;
  color: #b2dfdb;
  height: 50px;
  width: 100px;
  font-size: 18px;
  margin: 0 10px;
  -webkit-transition: background 0.3s;
  transition: background 0.3s;
}

.c0:hover {
  -webkit-transition: background 0.3s;
  transition: background 0.3s;
  background: #004d40;
}

<button
  className="c0"
/>
`;
